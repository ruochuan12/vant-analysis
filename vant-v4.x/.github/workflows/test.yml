name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - main

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v4

    - name: Install pnpm
      run: corepack enable

    - uses: actions/setup-node@v3
      with:
        node-version: '18'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: macos-14
    steps:
    - uses: actions/checkout@v4

    - name: Install pnpm
      run: corepack enable

    - uses: actions/setup-node@v3
      with:
        node-version: '18'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: pnpm run test:coverage

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v3
      with:
        token: ${{ secrets.CODECOV_TOKEN }}
        directory: ./packages/vant/test/coverage

  build:
    runs-on: macos-14
    steps:
    - uses: actions/checkout@v4

    - name: Install pnpm
      run: corepack enable

    - uses: actions/setup-node@v3
      with:
        node-version: '18'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
