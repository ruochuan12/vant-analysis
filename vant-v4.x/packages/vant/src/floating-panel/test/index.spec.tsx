import { later, mount, triggerDrag } from '../../../test';
import FloatingPanel from '..';

test('should minHeight 100 and maxHeight 0.6 innerHeight when anchors props do not', async () => {
  const wrapper = mount({
    render() {
      return <FloatingPanel>Content</FloatingPanel>;
    },
  });

  expect(wrapper.html()).toMatchSnapshot();

  expect((wrapper.element as HTMLDivElement).style.height).toBe(
    `${Math.round(window.innerHeight * 0.6)}px`,
  );

  expect((wrapper.element as HTMLDivElement).style.transform).toContain(
    '-100px',
  );
});

test('should drag adsorption effect when anchors props is [100, 200, 400]', async () => {
  const wrapper = mount({
    render() {
      return <FloatingPanel anchors={[100, 200, 400]}>内容</FloatingPanel>;
    },
  });

  expect(wrapper.html()).toMatchSnapshot();

  expect((wrapper.element as HTMLDivElement).style.height).toBe('400px');

  // drag 10
  await triggerDrag(wrapper.find('.van-floating-panel__header'), 0, 10);
  await later();
  expect((wrapper.element as HTMLDivElement).style.transform).toContain(
    '-100px',
  );

  expect((wrapper.element as HTMLDivElement).style.transform).toContain(
    '-100px',
  );

  // drag -49
  await triggerDrag(wrapper.find('.van-floating-panel__header'), 0, -49);
  await later();
  expect((wrapper.element as HTMLDivElement).style.transform).toContain(
    '-100px',
  );

  await triggerDrag(wrapper.find('.van-floating-panel__header'), 0, -199);
  await later();
  expect((wrapper.element as HTMLDivElement).style.transform).toContain(
    '-200px',
  );

  // drag -300
  await triggerDrag(wrapper.find('.van-floating-panel__header'), 0, -200);
  await later();
  expect((wrapper.element as HTMLDivElement).style.transform).toContain(
    '-400px',
  );

  // drag -500
  await triggerDrag(wrapper.find('.van-floating-panel__header'), 0, -500);
  await later();
  expect((wrapper.element as HTMLDivElement).style.transform).toContain(
    '-400px',
  );
});

test('should emit height-change when height change in anchors', async () => {
  const wrapper = mount({
    render() {
      return (
        <FloatingPanel
          anchors={[100, 200, 400]}
          onHeightChange={(h) => this.$emit('change', h)}
        >
          Content
        </FloatingPanel>
      );
    },
  });

  await triggerDrag(wrapper.find('.van-floating-panel__header'), 0, -199);
  await later();

  expect((wrapper.element as HTMLDivElement).style.transform).toContain(
    '-200px',
  );

  expect(wrapper.emitted('change')?.[0][0]).toEqual({ height: 200 });
});

test('should only drag header when allowDraggingContent is false', async () => {
  const wrapper = mount({
    render() {
      return (
        <FloatingPanel
          anchors={[100, 200, 400]}
          onHeightChange={(h) => this.$emit('change', h)}
          contentDraggable={false}
        >
          Content
        </FloatingPanel>
      );
    },
  });

  await triggerDrag(wrapper.find('.van-floating-panel__content'), 0, -199);
  await later();
  expect(wrapper.emitted('change')).toBeFalsy();

  await triggerDrag(wrapper.find('.van-floating-panel__header'), 0, -199);
  await later();
  expect(wrapper.emitted('change')).toBeTruthy();
});
