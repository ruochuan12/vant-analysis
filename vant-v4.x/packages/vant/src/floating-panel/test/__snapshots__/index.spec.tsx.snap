// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should drag adsorption effect when anchors props is [100, 200, 400] 1`] = `
<div
  class="van-floating-panel van-safe-area-bottom"
  style="height: 400px; transform: translateY(calc(100% + -100px)); transition: transform 0.3s cubic-bezier(0.18, 0.89, 0.32, 1.28);"
>
  <div class="van-floating-panel__header">
    <div class="van-floating-panel__header-bar">
    </div>
  </div>
  <div class="van-floating-panel__content">
    内容
  </div>
</div>
`;

exports[`should minHeight 100 and maxHeight 0.6 innerHeight when anchors props do not 1`] = `
<div
  class="van-floating-panel van-safe-area-bottom"
  style="height: 461px; transform: translateY(calc(100% + -100px)); transition: transform 0.3s cubic-bezier(0.18, 0.89, 0.32, 1.28);"
>
  <div class="van-floating-panel__header">
    <div class="van-floating-panel__header-bar">
    </div>
  </div>
  <div class="van-floating-panel__content">
    Content
  </div>
</div>
`;
